### PR TITLE
Remove headers and sections

### PR DESCRIPTION
**Pull request by eumiro**
Remove old headers from source code and possibly replace them with docstrings. Remove formatting comments (sometimes outdated) for sections in code.

### CODE REVIEW

File: tables/array.py [hunk 1]
```diff
@@ -92,17 +82,10 @@ class Array(hdf5extension.Array, Leaf):
     # Class identifier.
     _c_classid = 'ARRAY'
 
-    # Lazy read-only attributes
-    # `````````````````````````
-    @lazyattr
```
Comment (avalentino) [inline tables/array.py:97]: Is this wanted?

**Review by avalentino — CHANGES_REQUESTED** [review]

File: tables/array.py (hunk 1 above)
Comment (eumiro) [reply tables/array.py:97]: No, sorry, a silly typo, thank you, fixed, repushed. I hoped I was able to remove plenty of comments and not break anything, so I let it directly to the CI. And CI and you saw it.

**Comment by eumiro** [comment]
Finally a smaller PR where I can ask this:

1. I created this PR.
2. @avalentino `requested changes` by pointing to some line(s) in the diff and writing a comment
3. I replied in that mini-conversation, fixed my error, repushed (so those lines became `outdated`) and clicked on `Resolve conversation`.
4. But in the big final block, there's still the red text `Changes requested` with the field `1 change requested` and then it unrolls to `avalentino requested changes` with three dots and `See review`, `Dismiss review`, and `Re-request review`. Why doesn't this disappear if I `resolved` the conversation (subjectively decided that the change request has been processed) above and made the commented lines of code invalid anyway because I changed them in the force-push?

Just asking what is the official way to go in code review.

**Review by avalentino — APPROVED** [review]

**Comment by avalentino** [comment]
probably github is not able to determine that the requested change has been actually implemented.
Requesting a new revirew IMHO is the way to solve

**Pull request merged**